pkg/artifactory/resource_xray_policy_test.go

```diff
@@ -214,21 +214,23 @@ func testAccCheckPolicyDestroy(s *terraform.State) error {
 		}
 
 		policy, resp, err := conn.V1.Policies.GetPolicy(context.Background(), rs.Primary.ID)
-
-		if resp.StatusCode == http.StatusNotFound {
-			continue
-		} else if resp.StatusCode == http.StatusInternalServerError && err.Error() == fmt.Sprintf("{\"error\":\"Failed to find Policy %s\"}", rs.Primary.ID) {
-			continue
-		} else if err != nil {
-			return fmt.Errorf("error: Request failed: %s", err.Error())
+		if resp == nil {
```

Comment:
Check `err` first. If you check error first you probably never need to check if `resp` is `nil`.